### PR TITLE
add japanese ASR corpus CSJ to WeNet

### PR DESCRIPTION
**Pull request by Xianchao-Wu**
add japanese ASR corpus CSJ to WeNet.
With:
1. xml preprocessing and spliting;
2. wav preprocessing and spliting (by sentences, 2channel to 1channel)
3. text, wav.scp for wenet
4. sentencepiece with 4096 vocab size and bpe for tokenization
5. training
6. testing

### CODE REVIEW

File: examples/csj/s0/README.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+* Feature info: using fbank feature, cmvn, dither, online speed perturb
+* Training info: train_conformer.yaml, kernel size 15, lr 0.004, batch size 12, 8 gpu, acc_grad 1, 50 epochs, dither 0.0
+* Decoding info: ctc_weight 0.5, average_num 10
+
+
+| decoding mode                    | test1      | test2      | test3      |
+|----------------------------------|------------|------------|------------|
+| ctc greedy search                | 7.94       | 5.29       | 6.10       |
+| ctc prefix beam search           | 7.83+      | 5.28       | 6.08       |
```
Comment (robin1001) [inline examples/csj/s0/README.md:16]: what's the meaning of `+`?

File: examples/csj/s0/csj_tools/wn.2.prep.text.py [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+
+            wavabspath = os.path.abspath(wav)
+            aoutline = '{}\t{}\n'.format(afullid, wavabspath)
+            bw.write(aoutline)
+
+
+def prep_text_wavscp(
+        xmlpath, wavpath, test1fn, test2fn, test3fn,
+        outtrainpath, out1path, out2path, out3path):
+    # xmlpath = alike "/workspace/asr/wenet/examples/csj/s0/data/xml"
```
Comment (robin1001) [inline examples/csj/s0/csj_tools/wn.2.prep.text.py:117]: Remove the testing code?

File: examples/csj/s0/run.sh [hunk 3]
```diff
@@ -0,0 +1,346 @@
...
+average_num=10
+decode_modes="attention_rescoring ctc_greedy_search ctc_prefix_beam_search attention"
+
+. tools/parse_options.sh || exit 1;
+
+# bpemode (unigram or bpe)
+nbpe=4096 # TODO
+bpemode=bpe #unigram # TODO
+
+set -e # 如果任何语句的执行结果不是true，则bash退出
```
Comment (robin1001) [inline examples/csj/s0/run.sh]: use English comment

File: examples/csj/s0/run.sh [hunk 4]
```diff
@@ -0,0 +1,346 @@
...
+#stop_stage=4 #
+
+#stage=5 # sentence piece's bpe vocabulary
+#stop_stage=5 #
+
+#stage=6 # make "data.list" files
+#stop_stage=6 #
+
+#stage=7 # train -> 50 epochs
+#stop_stage=7 #
```
Comment (robin1001) [inline examples/csj/s0/run.sh]: remove the useless testing code.

File: examples/csj/s0/run.sh [hunk 5]
```diff
@@ -0,0 +1,346 @@
...
+      --val_best
+  fi
+  # Specify decoding_chunk_size if it's a unified dynamic chunk trained model
+  # -1 for full chunk
+  decoding_chunk_size=-1 # TODO for what?
+  ctc_weight=0.5
+  # Polling GPU id begin with index 0
+  num_gpus=$(echo $CUDA_VISIBLE_DEVICES | awk -F "," '{print NF}')
+  idx=0
+  #decode_modes1="ctc_greedy_search attention" # TODO better for batch_size=16 or 32
```
Comment (robin1001) [inline examples/csj/s0/run.sh]: remove the testing code

File: examples/csj/s0/run.sh [hunk 6]
```diff
@@ -0,0 +1,346 @@
...
+
+if [ ${stage} -le 6 ] && [ ${stop_stage} -ge 6 ]; then
+  # Export the best model you want
+  python wenet/bin/export_jit.py \
+    --config $dir/train.yaml \
+    --checkpoint $dir/avg_${average_num}.pt \
+    --output_file $dir/final.zip
+fi
+
+# Optionally, you can add LM and test it with runtime.
```
Comment (robin1001) [inline examples/csj/s0/run.sh]: We can remove the optional code here when LM is not supported.

File: examples/csj/s0/conf/train_conformer.yaml [hunk 7]
```diff
@@ -0,0 +1,80 @@
+# network architecture
```
Comment (robin1001) [inline examples/csj/s0/conf/train_conformer.yaml:1]: we can only checkin the basic training yaml(train_conformer.yaml) you used in this PR, and add other configures when you add new result  in the future.

so we can keep small PR which is perferred for both review and development

File: examples/csj/s0/csj_tools/wn.2.prep.text.py (hunk 2 above)
Comment (Xianchao-Wu) [reply examples/csj/s0/csj_tools/wn.2.prep.text.py]: sure! and done.

File: examples/csj/s0/run.sh (hunk 3 above)
Comment (Xianchao-Wu) [reply examples/csj/s0/run.sh]: changed to english comments

File: examples/csj/s0/run.sh (hunk 4 above)
Comment (Xianchao-Wu) [reply examples/csj/s0/run.sh]: removed

File: examples/csj/s0/run.sh (hunk 6 above)
Comment (Xianchao-Wu) [reply examples/csj/s0/run.sh:298]: sure

File: examples/csj/s0/csj_tools/wn.4.make_raw_list.py [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import json
+
+if __name__ == '__main__':
```
Comment (robin1001) [inline examples/csj/s0/csj_tools/wn.4.make_raw_list.py:20]: what's the difference to original `make_raw_list.py`

**Comment by Xianchao-Wu** [comment]
removed the debug code, change to english comments, remove not used config files and test output files

File: examples/csj/s0/csj_tools/wn.4.make_raw_list.py (hunk 8 above)
Comment (Xianchao-Wu) [reply examples/csj/s0/csj_tools/wn.4.make_raw_list.py:20]: removed the 1 to 1 mapping (between text and wav) in the the original make_raw_list.py -> only holds for csj dataset

File: examples/csj/s0/run.sh (hunk 5 above)
Comment (Xianchao-Wu) [reply examples/csj/s0/run.sh]: removed

File: examples/csj/s0/conf/train_conformer.yaml (hunk 7 above)
Comment (Xianchao-Wu) [reply examples/csj/s0/conf/train_conformer.yaml:1]: sure, only one yaml is left now

**Review by robin1001 — DISMISSED** [review]

**Review by robin1001 — APPROVED** [review]

**Pull request merged**